### PR TITLE
Prelease tidy and add is_ci to Diagnostics

### PR DESCRIPTION
**Pull request by Hoverbear**
##### Description

Fix use of a deprecated crate, update some deps, remove some dead code.

##### Checklist

- [ ] Formatted with `cargo fmt`
- [ ] Built with `nix build`
- [ ] Ran flake checks with `nix flake check`
- [ ] Added or updated relevant tests (leave unchecked if not applicable)
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] Linked to related issues (leave unchecked if not applicable)

##### Validating with `install.determinate.systems`

If a maintainer has added the `upload to s3` label to this PR, it will become available for installation via `install.determinate.systems`:

```shell
curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix/pr/$PR_NUMBER | sh -s -- install
```

### CODE REVIEW

File: src/settings.rs [hunk 1]
```diff
@@ -216,7 +216,7 @@ pub struct CommonSettings {
     ///     "os-version": "22.04.1 LTS (Jammy Jellyfish)",
     ///     "triple": "x86_64-unknown-linux-gnu",
     ///     "action": "Install",
-    ///     "status": "Success"
+    ///     "status": "Success"I
```
Comment (cole-h) [inline src/settings.rs:219]:
```suggestion
    ///     "status": "Success"
```

Comment (Hoverbear) [reply src/settings.rs:219]: Oh my goodness.

File: src/diagnostics.rs [hunk 2]
```diff
@@ -36,8 +35,11 @@ pub struct DiagnosticReport {
     pub os_name: String,
     pub os_version: String,
     pub triple: String,
+    pub is_ci: bool,
     pub action: DiagnosticAction,
     pub status: DiagnosticStatus,
+    /// Generally this includes the [`strum::IntoStaticStr`] representation of the error, we take special care not to include parameters of the error (which may include secrets)
+    pub variant: Option<String>,
```
Comment (cole-h) [inline src/diagnostics.rs:42]: We discussed renaming this so its meaning is more obvious when looking through the diagnostics:

```suggestion
    pub failure_variant: Option<String>,
```

**Review by cole-h — APPROVED** [review]
Aside from the above comment (which is not blocking, but something that we agreed should happen anyways), LGTM! :shipit:

**Pull request merged**